docs/docs/adrs/adr-018-permissionless-ics.md

```diff
@@ -0,0 +1,270 @@
+| `AllowlistPrefix`                        |Stores the allowlist of a chain                                                   |**YES**                   |Every parameter update                                                    |
+| `DenylistPrefix`                         |Stores the denylist of a chain                                                    |**YES**                   |Every parameter update                                                    |
+| `ConsumerRewardsAllocationBytePrefix`    |Stores the ICS rewards per chain                                                  |**YES**                   |Every IBC transfer packet that sends rewards to the provider              |
+| `ConsumerCommissionRatePrefix`           |Comission rate per chain per validator                                            |**YES**                   |Every `MsgSetConsumerCommissionRate` message                              |
+| `MinimumPowerInTopNBytePrefix`           |Stores the minimum power needed to opt in for a chain                             |**YES**                   |Every epoch                                                               |
+| `ConsumerAddrsToPruneV2BytePrefix`       |Stores consumer addresses to be pruned (as part of `VSCMaturedPacket`s deprecation)|**YES**                   |Every `MsgAssignConsumerKey` or `MsgOptIn` and later during actual pruning|
+
+Everything stored under a key associated with a `chainID` needs to be migrated to new state under `consumerID`. 
+Because we have to migrate in any case, we can also clean up a number of those keys by building a `ConsumerChainRecord`.
+The `ConsumerChainRecord` contains state relevant to a consumer chain and is keyed by a `consumerID`. Although the `ConsumerChainRecord`
+could contain **all** state related to a consumer chain (e.g., opted-in and consumer validators of a chain) we do not include
+such fields in the `ConsumerChainRecord` because this would increase the cost of ICS-related transactions due to the [gas cost](https://github.com/cosmos/cosmos-sdk/blob/v0.50.7/store/gaskv/store.go#L40).
+Furthermore, if we were to store all the opted-in or consumer validators, etc. it would be tricky to read or write those fields,
+because we would need to extract the opted-in validators from the `ConsumerChainRecord` and then search through those
+validators to find the one we are looking for, etc.
+
+The `ConsumerChainRecord` contains the following:
+```protobuf
+message ConsumerChainRecord {
+  // the owner of this consumer chain
+  string owner_address;
+  // client id to the consumer chain
+  string client_id;
+  // channel id
+  string channel_id;
+  // the chain-id of the consumer chain
+  string chain_id;
+  // consumer genesis
+  interchain_security.ccv.v1.ConsumerGenesisState consumer_genesis = 5
+  // min height needed for an equivocation to be actionable
```

Comment:
at first, I thought this would continuously update.
In reality, I think this is just the first height the chain is ICS secured for? Could the comment be a bit clearer on this?

Reply:
Yes, that is correct. Rephrased.